hack/.ci/run-e2e-gke.sh

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env bash
+#
+# Copyright (C) 2023 ScyllaDB
+#
+
+set -euExo pipefail
+shopt -s inherit_errexit
+
+if [ -z ${SO_SUITE+x} ]; then
+  echo "SO_SUITE can't be empty" > /dev/stderr
+  exit 2
+fi
+
+if [ -z ${SO_IMAGE+x} ]; then
+  echo "SO_IMAGE can't be empty" > /dev/stderr
+  exit 2
+fi
+
+SO_DISABLE_NODECONFIG=${SO_DISABLE_NODECONFIG:-false}
+
+field_manager=hack
+
+function kubectl_create {
+    if [[ -z ${REENTRANT+x} ]]; then
+        # In an actual CI run we have to enforce that no two objects have the same name.
+        kubectl create --field-manager="${field_manager}" "$@"
+    else
+        # For development iterations we want to update the objects.
+        kubectl apply --server-side=true --field-manager="${field_manager}" --force-conflicts "$@"
+    fi
+}
+
+function gather-artifacts {
+  ARTIFACTS_DIR="${ARTIFACTS}" timeout -v 50m ./hack/ci-gather-artifacts.sh
+  # Events are costly to gather, so we'll erase them for now. (e2e namespaces are collected separately)
+  kubectl delete --raw /api/v1/namespaces/default/events || true
+  kubectl delete --raw /api/v1/namespaces/kube-system/events || true
+}
+
+function handle-exit {
+  gather-artifacts || "Error gathering artifacts" > /dev/stderr
+}
+
+trap handle-exit EXIT
+
+# Allow admin to use ephemeralcontainers
+kubectl_create -f - <<EOF
+  apiVersion: rbac.authorization.k8s.io/v1
+  kind: ClusterRole
+  metadata:
+    name: scylladb-e2e:hotfixes
+    labels:
+      rbac.authorization.k8s.io/aggregate-to-admin: "true"
+  rules:
+  - apiGroups:
+    - ""
+    resources:
+    - pods/ephemeralcontainers
+    verbs:
+    - patch
+EOF
+
+kubectl apply --server-side -f ./pkg/api/scylla/v1alpha1/scylla.scylladb.com_nodeconfigs.yaml
+kubectl wait --for condition=established crd/nodeconfigs.scylla.scylladb.com
+
+if [[ "${SO_DISABLE_NODECONFIG}" == "true" ]]; then
+  echo "Skipping NodeConfig creation"
+else
+  kubectl_create -f - << EOF
+apiVersion: scylla.scylladb.com/v1alpha1
```

Comment:
nit: NodeConfig could also be moved to a manifest file

Reply:
done